runtime/lib/capi/RuntimeCAPI.cpp

```diff
@@ -145,7 +145,7 @@ void __quantum__qis__Gradient(int64_t numResults, /* results = */...)
     // num_observables * num_train_params
     auto &&jacobian = Catalyst::Runtime::CAPI::get_device()->Gradient({});
 
-    const size_t num_observables = jacobian.size();
+    const auto num_observables = jacobian.size();
```

Comment:
I don't know if this is contentious or not, but is it really preferable to remove all type information from the source code?

Sometimes the type is immediately because of a cast, or sometimes more convenient when the type is super unwieldy, but what do we gain from erasing all types in the source?